### DISCUSSION TITLE
Stop and remove a project with a customer agreement active

### DISCUSSION THREAD

**Discussion by yasiryagi** [subquery/subql#965] [Ideas]
When trying to stop indexing a project with a customer agreement active, I get the error below. 
Also the GUI become a bit confusing. So it reject the request and does not go on chain, but then the remove project works and the project disappear but when you add the project back it comes with the same state before removal.

> [Reported issue 917] (https://github.com/subquery/subql/discussions/917)
```
coordinator_proxy    | unhandled error: Rejection(QueryError("error sending request for url (http://query_qmyr8xqgaxucxmp:3001/): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution"))
coordinator_proxy    | unhandled error: Rejection(QueryError("error sending request for url (http://query_qmyr8xqgaxucxmp:3001/): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution"))
coordinator_proxy    | unhandled error: Rejection(QueryError("error sending request for url (http://query_qmyr8xqgaxucxmp:3001/): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution"))
coordinator_db       | 2022-05-04 03:12:46.162 UTC [2885] LOG:  unexpected EOF on client connection with an open transaction
```

![8](https://user-images.githubusercontent.com/4862448/166652713-5e34ed96-0b4c-4369-b38a-fdb058e88b35.PNG)

![9](https://user-images.githubusercontent.com/4862448/166652736-1e1821c2-2a71-4fac-9300-187b6bd20995.PNG)
![10](https://user-images.githubusercontent.com/4862448/166652748-bc3251a0-06bd-41ea-a55c-50e733eb4a04.PNG)
![11](https://user-images.githubusercontent.com/4862448/166652757-8dc83b5f-e6d8-44d1-91bd-e258defc25cc.PNG)
![12](https://user-images.githubusercontent.com/4862448/166652772-5f6de1b8-e55e-45ed-b6f1-000aab2b94e2.PNG)

**Comment by mzxyz** [comment]
Thanks for reporting this.

1. Indexer can not Announce `NOT INDEXING` with ongoing service agreements.

2. Once finish the first step for stopping indexer, the query/node containers will be terminated. At this stage, indexer can choose to restart indexing the project with different configs, or remove the project.

3. The indexing status `READY` is a status on blockchain, you can only modify it by sending a transaction. The `PROJECT STATUS` is the status of you running project. So when you add the project back, the network status is still `READY`, but the project is not indexing, the inconsistent status should be managed by indexers as if there has ongoing service agreements for a project but indexer make the service offline (terminated), will got potential punishment from the network.

4. Back to this issue, if the reason for stopping  indexing is to upgrade the service versions to the latest one, you don't need to send the transaction for announce not indexing. 
<img width="1377" alt="Screen Shot 2022-05-04 at 9 25 31 PM" src="https://user-images.githubusercontent.com/8177474/166656843-49914871-03a6-4cf7-bae0-1a2432fbc113.png">

5. We will remove the second step (send transaction) when `stop indexing` to make it a pure action to stop the running project. Indexer can manually send the transaction for announce not indexing.

**Comment by yasiryagi** [comment]
The confusion came from the ability to remove the project while the indexing halting is rejected. One should should just get a rejection message and cannot go further. 

Can we disable the ability to remove the project while indexing is not fully stopped.

**Comment by yasiryagi** [comment]
I prefer if we just make a clear message and min UI changes. What confuses is that the UI changes while one not sure what actions happened. We should do:
- Take a way the ability to remove the project while the project is active.
- - Since the indexing will be stopped after the service agreement expires, message that clearly in the main UI with min changes (Not talking about the error message).
- Make minimum changes to the UI after action rejection.
